kube/src/api/resource.rs

```diff
@@ -245,9 +156,9 @@ impl Resource {
     /// Replace an instance of a resource
     ///
     /// Requires `metadata.resourceVersion` set in data
-    pub fn replace(&self, name: &str, pp: &PostParams, data: Vec<u8>) -> Result<http::Request<Vec<u8>>> {
-        let base_url = self.make_url() + "/" + name + "?";
-        let mut qp = url::form_urlencoded::Serializer::new(base_url);
+    pub fn replace(self, name: &str, pp: &PostParams, data: Vec<u8>) -> Result<http::Request<Vec<u8>>> {
```

Comment:
`patch` is inconsistent with the rest. Keep `&self`?

Reply:
ah, good spot. i meant to take ownership everywhere in `Request` actually, although it's not technically needed here.

Reply:
Actually, it's a lot nicer to use `&self` as you suggested. Means we have to use one extra `format!` inside `Request`, but it means that `Api` can store a `Request` with the `path_url`, and not have to worry about the family or namespace again.

Reply:
That actually cleaned up quite a bit. DynamicType no longer needs to be Clone within kube as a result of reference use of Request. Also means less  string creations (if people use the Api for more than one Request).